### PR TITLE
[tflite] implement serialize

### PR DESCRIPTION
**Pull request by zhoonit**
- [tflite] implement serialize 

```
This patch implements serialize with bunch of NYI functions, structures

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

File: nntrainer/compiler/tflite_interpreter.cpp [hunk 1]
```diff
@@ -12,14 +12,21 @@
...
 #include <nntrainer_error.h>
+#include <node_exporter.h>
+#include <tensor.h>
+
+#define UNUSED(x) x __attribute__((unused))
+#define WILL(x) ;
 
 static constexpr const char *FUNC_TAG = "[TFLITE INTERPRETER] ";
 
+namespace nntrainer {
```
Comment (jijoongmoon) [inline nntrainer/compiler/tflite_interpreter.cpp:29]: is it intended? ```namespace nntrainer{ namespace{ ```??

File: nntrainer/compiler/tflite_interpreter.cpp [hunk 2]
```diff
@@ -42,32 +49,110 @@ void builder2file(const flatbuffers::FlatBufferBuilder &builder,
...
-  /// 1. in&outs&weights
-  /// 2. buffer
-  /// 2. ops
-  /// 3. op_codes
+  auto opNodes = buildOpNodes(representation);
+  TfOpIdxMap map(opNodes); /// build TfOpIdxMap
 
-  auto desc = fb_builder.CreateString("This file is generated from NNTrainer");
+  auto UNUSED(opcodes) = buildOperatorCodes(map, fbb);
+  auto UNUSED(buffers) = buildBuffers(map, fbb);
```
Comment (jijoongmoon) [inline nntrainer/compiler/tflite_interpreter.cpp:141]: is this just for temporal? or intended?

File: nntrainer/compiler/tflite_interpreter.cpp [hunk 3]
```diff
@@ -42,32 +49,110 @@ void builder2file(const flatbuffers::FlatBufferBuilder &builder,
...
-} // namespace
 
-namespace nntrainer {
+/**
+ * @brief tensorflow operational node representation. This class contains,
+ * information to build operation flatbuffer
+ *
+ */
+class TfOpNode {
+private:
```
Comment (jijoongmoon) [inline nntrainer/compiler/tflite_interpreter.cpp:59]: is it layer level?

File: nntrainer/compiler/tflite_interpreter.cpp [hunk 4]
```diff
@@ -42,32 +49,110 @@ void builder2file(const flatbuffers::FlatBufferBuilder &builder,
...
+flatbuffers::Offset<
+  flatbuffers::Vector<flatbuffers::Offset<tflite::OperatorCode>>>
+buildOperatorCodes(const TfOpIdxMap &map, flatbuffers::FlatBufferBuilder &fbb) {
+  /** NYI! */
+  return flatbuffers::Offset<
+    flatbuffers::Vector<flatbuffers::Offset<tflite::OperatorCode>>>();
+};
+
+flatbuffers::Offset<flatbuffers::Vector<flatbuffers::Offset<tflite::SubGraph>>>
+buildSubGraph(const TfOpNodes &nodes, const TfOpIdxMap &map,
```
Comment (jijoongmoon) [inline nntrainer/compiler/tflite_interpreter.cpp:122]: I wonder what is the SubGraph?

File: nntrainer/compiler/tflite_interpreter.cpp (hunk 1 above)
Comment (zhoonit) [reply nntrainer/compiler/tflite_interpreter.cpp:29]: Yes, those functions are not used outside of this file.

File: nntrainer/compiler/tflite_interpreter.cpp (hunk 2 above)
Comment (zhoonit) [reply nntrainer/compiler/tflite_interpreter.cpp:141]: Yes, this is to show the intention of this function. this is to disable compiler warning.

File: nntrainer/compiler/tflite_interpreter.cpp (hunk 3 above)
Comment (zhoonit) [reply nntrainer/compiler/tflite_interpreter.cpp:59]: Yes, this corresponds to each layer.

File: nntrainer/compiler/tflite_interpreter.cpp (hunk 4 above)
Comment (zhoonit) [reply nntrainer/compiler/tflite_interpreter.cpp:122]: subGraph is a concept from tflite file. It is basically same as our graph.

File: nntrainer/compiler/tflite_interpreter.cpp [hunk 5]
```diff
@@ -42,32 +49,110 @@ void builder2file(const flatbuffers::FlatBufferBuilder &builder,
...
+
+private:
+  /**
+   * @brief Bidirectional Index map
+   *
+   * @tparam T type of a underyling value
+   */
+  template <typename T> class BidirectionalIndexMap {
+    std::unordered_map<T, unsigned int> data2index; /**< data -> index map */
+    std::vector<T> index2data;                      /**< index -> data map */
```
[the remark this answers is not in the document]
Comment (zhoonit) [reply nntrainer/compiler/tflite_interpreter.cpp:90]: Yes, they are.

**Review by kparichay — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM. ( UNUSED will be removed in the end )

**Pull request merged**